File: Pythonleetcode/Leetcode136.py
# Approach 1
# class Solution(object):
#     def singleNumber(self, nums):
#         nums.sort()
#         if len(nums)==1:
#             return nums[0]
#         if nums[0]!=nums[1]:
#             return nums[0]
#         if nums[len(nums)-1]!=nums[len(nums)-2]:
#             return nums[len(nums)-1]
#
#         for i in range(1,len(nums)-1):
#             if nums[i]!=nums[i-1] and nums[i]!=nums[i+1]:
#                 return nums[i]

#Approach 2
class Solution(object):
    def singleNumber(self, nums):
        dict={}
        for i in nums:
            dict[i]= dict.get(i,0)+1
        for i in dict:
            if dict[i]==1:
                return i


print(Solution().singleNumber([1,2,4,1,2]))

File: CppPractice/Project1/Project1/Leetcode136.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


class Solution {
public:
    int singleNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        if (nums.size()==1)
        {
            return nums[0];
        }
        for (int i = 1; i < nums.size()-1; i++)
        {
            if (nums[i]!=nums[i-1] and nums[i]!=nums[i+1])
            {
                return nums[i];
            }
        }
        if (nums[0]==nums[1])
        {
            return nums[nums.size() - 1];
        }
        else
        {
            return nums[0];
        }
    }
};